src/metal.cs

```diff
@@ -2050,5 +2399,242 @@ interface MTLStageInputOutputDescriptor : NSCopying
+	[BaseType (typeof(MTLType))]
+	interface MTLTextureReferenceType
+	{
+		[Export ("textureDataType")]
+		MTLDataType TextureDataType { get; }
+
+		[Export ("textureType")]
+		MTLTextureType TextureType { get; }
+
+		[Export ("access")]
+		MTLArgumentAccess Access { get; }
+
+		[Export ("isDepthTexture")]
+		bool IsDepthTexture { get; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	interface IMTLCaptureScope { }
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface MTLCaptureScope
+	{
+		[Abstract]
+		[Export ("beginScope")]
+		void BeginScope ();
+
+		[Abstract]
+		[Export ("endScope")]
+		void EndScope ();
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+
+		[Abstract]
+		[Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("commandQueue")]
+		IMTLCommandQueue CommandQueue { get; }
+	}
+
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface MTLCaptureManager
+	{
+		[Static]
+		[Export ("sharedCaptureManager")]
+		MTLCaptureManager SharedCaptureManager { get; }
+
+		[Export ("newCaptureScopeWithDevice:")]
+		IMTLCaptureScope CreateNewCaptureScope (IMTLDevice device);
+
+		[Export ("newCaptureScopeWithCommandQueue:")]
+		IMTLCaptureScope CreateNewCaptureScope (IMTLCommandQueue commandQueue);
+
+		[Export ("startCaptureWithDevice:")]
+		void StartCapture (IMTLDevice device);
+
+		[Export ("startCaptureWithCommandQueue:")]
+		void StartCapture (IMTLCommandQueue commandQueue);
+
+		[Export ("startCaptureWithScope:")]
+		void StartCapture (IMTLCaptureScope captureScope);
+
+		[Export ("stopCapture")]
+		void StopCapture ();
+
+		[NullAllowed, Export ("defaultCaptureScope", ArgumentSemantic.Strong)]
+		IMTLCaptureScope DefaultCaptureScope { get; set; }
+
+		[Export ("isCapturing")]
+		bool IsCapturing { get; }
+	}
+
+	[Mac (10, 13, onlyOn64: true), NoiOS, NoTV, NoWatch]
+	enum MTLDeviceNotificationName {
+		[Field ("MTLDeviceWasAddedNotification")]
+		DeviceWasAdded,
+
+		[Field ("MTLDeviceRemovalRequestedNotification")]
+		DeviceRemovalRequested,
+
+		[Field ("MTLDeviceWasRemovedNotification")]
+		MTLDeviceWasRemoved,
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPipelineBufferDescriptor : NSCopying
+	{
+		[Export ("mutability", ArgumentSemantic.Assign)]
+		MTLMutability Mutability { get; set; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPipelineBufferDescriptorArray
+	{
+		[Export ("objectAtIndexedSubscript:")]
+		MTLPipelineBufferDescriptor GetObject (nuint bufferIndex);
+
+		[Export ("setObject:atIndexedSubscript:")]
+		void SetObject ([NullAllowed] MTLPipelineBufferDescriptor buffer, nuint bufferIndex);
```

Comment:
I think both could be `[Protected]` and we could expose an indexer (in manual bindings) ?